staging/src/k8s.io/api/extensions/v1beta1/types.go

```diff
@@ -569,6 +569,19 @@ type IngressList struct {
 
 // IngressSpec describes the Ingress the user wishes to exist.
 type IngressSpec struct {
+	// IngressClassName is the name of the IngressClass cluster resource. The
+	// associated IngressClass defines which controller will implement the
+	// resource. This replaces the deprecated `kubernetes.io/ingress.class`
+	// annotation. For backwards compatibility, when that annotation is set, it
+	// must be given precedence over this field. The controller may emit a
+	// warning if the field and annotation have different values.
+	// Implementations of this API may choose to ignore Ingresses without a
+	// class specified. An IngressClass resource may be marked as default, which
```

Comment:
Does that mean that unsetting an ingressClassName is unexpected? Old clients have the potential to drop the ingressClassName field... should we preserve it for them in `ingressStrategy#PrepareForUpdate`?

Reply:
It is likely going to be rare, but I don't know if we want to completely prevent it from being unset. For example, an end user may want to switch to an Ingress controller that does not yet support the class field, so they may want to unset the field and set the annotation.